### DISCUSSION TITLE
Did  the rook version 1.2.7 supported the kuberbetes version  1.19,1.20,1.21,1.22,1.23?

### DISCUSSION THREAD

**Discussion by zhangxingdeppon** [rook/rook#9402] [Q&A]
**Is this a bug report or feature request?**

Hi, team, I have a doubt about the rook version 1.2.7, did  the rook version 1.2.7 supported the kuberbetes version  1.19,1.20,1.21,1.22,1.23?, I have seen the office document `quick start`, just describe:
```
Kubernetes v1.11 or higher is supported by Rook.
```
**What is use case behind this feature:**

**Environment**:

**Comment by leseb** [comment]
Rook 1.2.7 is getting rather old so the best way would be to test it. However, I believe Flex is deprecated in Kubernetes upstream in 1.2X versions. So I'd suggest updating to the latest Rook version. Thanks

**Comment by travisn** [comment]
@zhangxingdeppon The supported K8s versions are really k8s 1.11 and newer *as of the data when that rook version was released*, so the newer versions of K8s may not work properly. If you have a newer version of K8s, it's recommended to have a newer version of Rook!

**Comment by zhangxingdeppon** [comment]
@travisn @leseb happy to receive feedback,  the main reason is that our services rely on rook version 1.2.7 for ceph, we need to use ceph filestore on the newer versions of K8s, I known it would better to use a newer version of Rooks, but the rook version 1.3.0 declared `Support for directory-based (FileStore) OSDs is removed. Raw partitions are supported as an alternative`, it means filestore was removed after 1.3.0 , and we can't upgrade it yet because of the infrastructure, so maybe it's a good idea to continue to use the rook 1.2.7,  for the above situation, could you give me some ideas? it's important for me , thanks very much.

**Reply by travisn** [reply]
The main challenge with Rook 1.2.7 on newer K8s versions is that in K8s v1.22, support for the [CRDs with version apiextensions.k8s.io/v1beta1 was removed](https://kubernetes.io/docs/reference/using-api/deprecation-guide/#customresourcedefinition-v122). This likely means K8s 1.21 is the final version you can use with Rook v1.2.7. Really, you will need to provision partitions or devices in your infrastructure to use rook on newer versions.

**Reply by zhangxingdeppon** [reply]
@travisn excellent, it helps a lot, according to your says, we decide to use the rook version 1.2.7 on kubernetes 1.20, thanks very much.
